### PR TITLE
CAMEL-14805: Component - Use doInit for wiring instead of doStart

### PR DESCRIPTION
**Pull request by zhfeng**
https://issues.apache.org/jira/browse/CAMEL-14805 
   * camel-bean
   * camel-dataformat
   * camel-dozer
   * camel-file
   * camel-sql
   * camel-stream
   * camel-timer
   * camel-xslt

### CODE REVIEW

File: components/camel-stream/src/main/java/org/apache/camel/component/stream/StreamEndpoint.java [hunk 1]
```diff
@@ -303,7 +303,8 @@ public void setReadTimeout(int readTimeout) {
     //-------------------------------------------------------------------------
 
     @Override
-    protected void doStart() throws Exception {
+    protected void doInit() throws Exception {
+        super.doInit();
```
Comment (davsclaus) [inline components/camel-stream/src/main/java/org/apache/camel/component/stream/StreamEndpoint.java:307]: charsets are potentially not available if native compiling, so we may consider to make this doStart. However there are ways to make native compilation include all charsets.

File: components/camel-sql/src/main/java/org/apache/camel/component/sql/stored/SqlStoredProducer.java [hunk 2]
```diff
@@ -123,8 +123,8 @@ private StatementWrapper createStatement(Exchange exchange) throws SQLException
     }
 
     @Override
-    protected void doStart() throws Exception {
-        super.doStart();
+    protected void doInit() throws Exception {
+        super.doInit();
```
Comment (davsclaus) [inline components/camel-sql/src/main/java/org/apache/camel/component/sql/stored/SqlStoredProducer.java:127]: This should be doStart as SqlHelper.resolveQuery can potentially load the SQL template from file system. See the SQL docs

File: components/camel-sql/src/main/java/org/apache/camel/component/sql/SqlProducer.java [hunk 3]
```diff
@@ -69,8 +69,8 @@ public SqlEndpoint getEndpoint() {
     }
 
     @Override
-    protected void doStart() throws Exception {
-        super.doStart();
+    protected void doInit() throws Exception {
+        super.doInit();
```
Comment (davsclaus) [inline components/camel-sql/src/main/java/org/apache/camel/component/sql/SqlProducer.java:73]: This should be doStart as SqlHelper.resolveQuery can potentially load the SQL template from file system. See the SQL docs

File: components/camel-sql/src/main/java/org/apache/camel/component/sql/SqlConsumer.java [hunk 4]
```diff
@@ -101,8 +101,8 @@ public DefaultSqlEndpoint getEndpoint() {
     }
 
     @Override
-    protected void doStart() throws Exception {
-        super.doStart();
+    protected void doInit() throws Exception {
+        super.doInit();
```
Comment (davsclaus) [inline components/camel-sql/src/main/java/org/apache/camel/component/sql/SqlConsumer.java:105]: This should be doStart as SqlHelper.resolveQuery can potentially load the SQL template from file system. See the SQL docs

**Comment by davsclaus** [comment]
I created ticket
https://issues.apache.org/jira/browse/CAMEL-15110

Then we can use that to know if the template/query for sql components etc will be loaded from classpath or not. If classpath you can doInit otherwise its in doStart.

**Comment by zhfeng** [comment]
Thanks @davsclaus and it's simply to test the resourceURI is started with "classpath://" ?

**Comment by davsclaus** [comment]
Yeah see this PR
https://github.com/apache/camel/commit/3e17141d88983a0830dc21135bec536a53513b6b

**Comment by zhfeng** [comment]
Cool ! I will update to use ```isClasspathUri```

**Comment by davsclaus** [comment]
Thanks @zhfeng let us know when you this PR is ready

**Pull request merged**